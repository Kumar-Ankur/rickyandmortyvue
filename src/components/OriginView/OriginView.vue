<template>
  <div class="species-content">
    <h2>{{ ORIGIN_FILTER }}</h2>

    <app-check-box
      v-for="(data, index) in originData"
      :key="index"
      :data="data"
      :addFilter="addFilter"
      :removeFilter="removeFilter"
      :modelName="modelName"
      :store="store"
    ></app-check-box>
  </div>
</template>

<script>
import CheckBox from "../CheckBox/CheckBox.vue";
import * as constants from "../../constants";
import { getCheckBoxData } from "../../utils";
export default {
  data() {
    return {
      ORIGIN: constants.ORIGIN,
      ORIGIN_FILTER: constants.ORIGIN_FILTER,
      originData: [],
      modelName: 'origin',
      addFilter: this.$store.dispatch("addOriginFilteredData"),
      removeFilter: this.$store.dispatch("removeOriginFilteredData"),
      store: this.$store
    };
  },
  components: {
    appCheckBox: CheckBox
  },
  computed: {
    characters() {
      return getCheckBoxData(
        this.$store.getters.characters,
        constants.ORIGIN,
        true
      );
    }
  },
  watch: {
    characters(newVal, oldVal) {
      this.originData = newVal;
    }
  }
};
</script>

<style scoped>
.species-content {
  color: white;
  border: 1px solid white;
  padding: 2rem;
  margin-left: 2rem;
}
</style>