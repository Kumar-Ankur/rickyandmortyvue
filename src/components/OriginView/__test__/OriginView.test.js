import { shallowMount } from '@vue/test-utils';
import OriginView from '../OriginView.vue';

test('generate snapshot', () => {
    const wrapper = shallowMount(OriginView, {
        mocks: {
            $store: {
                getters: {
                    characters: []
                }
            }
        }
    })
    expect(wrapper.html()).toMatchSnapshot();
})