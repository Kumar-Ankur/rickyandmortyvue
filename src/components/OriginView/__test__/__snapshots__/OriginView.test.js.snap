// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate snapshot 1`] = `
"<div class=\\"species-content\\">
  <h2>Origin Filter:</h2>
</div>"
`;
